eclipse-scout-core/src/table/Table.ts

```diff
@@ -5858,15 +5851,170 @@ export class Table extends Widget implements TableModel, Filterable<TableRow> {
+      .filter(column => !!column);
+
+    const columnsOld = [...this.columns];
+    const columnsDeleted = this.columns.filter(column => column instanceof Column && !arrays.contains(columns, column));
+    const columnsInserted = columns.filter(column => !arrays.contains(this.columns, column));
+    const columnsCurrent = [...this.columns];
+
+    let buffering: JQuery.Deferred<void>;
+
+    // deleted columns
+    for (const column of columnsDeleted) {
+      // update index of other columns and handle cells if removed column has an index set
+      if (column.index >= 0) {
+        // update index of all columns whose index is greater than the index of the removed column
+        for (const c of columnsCurrent) {
+          if (c.index > column.index) {
+            c.index--;
+          }
+        }
+
+        // remove cells for all rows at the correct position (column.index)
+        this.rows.forEach(row => row.cells.splice(column.index, 1));
+
+      }
+
+      // destroy column and remove from current columns
+      column.destroy();
+      arrays.remove(columnsCurrent, column);
+    }
+
+    // inserted columns
+    for (const column of columnsInserted) {
+      // set index to max index + 1 if no index is provided
+      if (column.index < 0) {
+        column.index = Math.max(-1, ...columnsCurrent.map(column => column.index)) + 1;
+      }
+
+      // update index of all columns whose index is greater or equal to the index of the new column
+      for (const c of columnsCurrent) {
+        if (c.index >= column.index) {
+          c.index++;
+        }
+      }
+
+      // set updateBuffer buffering until all columns are added, otherwise column.initCell triggers updateRow before the rows cell array is modified
+      if (!buffering) {
+        buffering = $.Deferred();
+        this.updateBuffer.pushPromise(buffering.promise());
+      }
+
+      // insert new cell for all rows at the correct position (column.index)
+      this.rows.forEach(row => {
+        arrays.insert(row.cells, column.initCell(initValue(column, row), row), column.index);
+      });
+
+      // add to current columns
+      columnsCurrent.push(column);
+    }
+
+    this._setProperty('columns', columns);
+
+    // resolve buffering and trigger events
+    buffering?.resolve();
+    this._triggerColumnStructureChanged(columnsOld, this.columns);
+  }
+
+  protected _renderColumns() {
+    this.columnLayoutDirty = true;
+    this._updateRowWidth();
+    this.redraw();
+  }
+
+  protected _triggerColumnStructureChanged(columnsOld: Column<any>[], columnsNew: Column<any>[]) {
+    // create new arrays to prevent unexpected behaviour if listeners modify the arrays
+    columnsOld = [...columnsOld];
+    columnsNew = [...columnsNew];
+    this.trigger('columnStructureChanged', {columnsOld, columnsNew});
+  }
+
+  insertColumn(columnOrModel: ObjectOrChildModel<Column<any>>, position?: number, initValue?: any | ((Column, TableRow) => any)) {
```

Comment:
It would be convenient if we could also pass a reference column instead of the position. That way, the caller would not need to compute the index by itself.

```suggestion
  insertColumn(columnOrModel: ObjectOrChildModel<Column<any>>, positionOrInsertAfterColumn?: number | Column<any>, initValue?: any | ((Column, TableRow) => any)) {
```